### PR TITLE
Terminate CLI port properly

### PR DESCRIPTION
**Pull request by hahnbeelee**
## Summary
Apparently using `spawn` is better for terminating ports properly than `exec` ([stackoverflow](https://stackoverflow.com/questions/20187184/how-to-kill-childprocess-in-nodejs)).

## Test Plan
1. Run `mintlify dev`
2. Terminate it
3. Run `lsof -i:3000`. There should be no node processes running. I think it's ok if the Google runs are running.
image

### CODE REVIEW

File: packages/mintlify/src/local-preview/index.ts [hunk 1]
```diff
@@ -168,9 +169,28 @@ const run = (port: string) => {
...
+      PORT: port,
+    },
+    cwd: CLIENT_PATH,
+    stdio: "pipe",
+    shell: true,
+  });
+  mintlifyDevProcess.stdout.on("data", (data) => {
+    const output = data.toString();
+    console.log(output);
+    if (output.startsWith("> Ready on http://localhost:")) {
```
Comment (tobias-mintlify) [inline packages/mintlify/src/local-preview/index.ts:184]: Can we move these logs into here so they only show when the site is ready?

```
console.log(
    `🌿 ${Chalk.green(
      `Your local preview is available at http://localhost:${port}`
    )}`
  );
  console.log(
    `🌿 ${Chalk.green("Press Ctrl+C any time to stop the local preview.")}`
  );
```

Comment (tobias-mintlify) [reply packages/mintlify/src/local-preview/index.ts:184]: Also change `logger.succeed("Local Mintlify instance started.");` above to `Local Mintlify instance is ready. Launching your site now.`

And `Starting a local Mintlify instance...` to `Preparing local Mintlify instance.`

**Review by tobias-mintlify — CHANGES_REQUESTED** [review]

File: packages/mintlify/src/local-preview/index.ts [hunk 2]
```diff
@@ -151,26 +152,45 @@ const dev = async (argv: ArgumentsCamelCase) => {
...
```
Comment (benjamin-shen) [inline packages/mintlify/src/local-preview/index.ts:176]: This is kind of a sus check, is there some env variable or something we could check instead?

**Review by benjamin-shen — APPROVED** [review]

File: packages/mintlify/src/local-preview/index.ts (hunk 2 above)
Comment (hahnbeelee) [reply packages/mintlify/src/local-preview/index.ts:176]: Oo maybe `lsof -i:3000` and check if it's running?

**Review by tobias-mintlify — APPROVED** [review]

**Pull request merged**